Doxygen documentation build fails

Today's nightly build of doxygen documentation fails with the following error:

> [320/321] Generating html Sphinx documentation for clang into "/home/runner/work/llvm/llvm/build/tools/clang/docs/html"
> FAILED: tools/clang/docs/CMakeFiles/docs-clang-html /home/runner/work/llvm/llvm/build/tools/clang/docs/CMakeFiles/docs-clang-html 
> cd /home/runner/work/llvm/llvm/build/tools/clang/docs && /usr/local/bin/cmake -E env /usr/local/bin/sphinx-build -b html -d /home/runner/work/llvm/llvm/build/tools/clang/docs/_doctrees-clang-html -q -t builder-html -W /home/runner/work/llvm/llvm/build/tools/clang/docs /home/runner/work/llvm/llvm/build/tools/clang/docs/html
> 
> Configuration error:
> There is a programmable error in your configuration file:
> 
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.8/dist-packages/sphinx/config.py", line 327, in eval_config_file
>     exec(code, namespace)
>   File "/home/runner/work/llvm/llvm/build/tools/clang/docs/conf.py", line 41, in <module>
>     import recommonmark
> ModuleNotFoundError: No module named 'recommonmark'
> 
> ninja: build stopped: subcommand failed.
> Error: Process completed with exit code 1.

I don't see much changes in documentation, so I guess it might be caused by llorg changes pulled with https://github.com/intel/llvm/commit/5c9f7c384bf333abbd55d68b1fe957252fe73cb0.
@vmaksimo, could you take a look, please?

--- a/.github/workflows/gh_pages.yml
+++ b/.github/workflows/gh_pages.yml
@@ -34,22 +34,6 @@ jobs:
         cmake --build . --target doxygen-clang
         cmake --build . --target docs-sycl-html
         cmake --build . --target docs-clang-html
-        cd $GITHUB_WORKSPACE
-        mkdir rendered
-        cd rendered
-        yes | \cp -rf $GITHUB_WORKSPACE/build/tools/sycl/doc/html/* .
-        mv $GITHUB_WORKSPACE/build/tools/sycl/doc/doxygen/html doxygen/
-        mv $GITHUB_WORKSPACE/build/tools/clang/docs/html clang/
-        mv $GITHUB_WORKSPACE/build/tools/clang/docs/doxygen/html clang_doxygen/
-    - name: Pack
-      if: ${{ github.event_name != 'schedule' }}
-      run: tar -czf sycl_docs.tar.gz -C $GITHUB_WORKSPACE/rendered .
-    - name: Upload docs as artifact
-      if: ${{ github.event_name != 'schedule' }}
-      uses: actions/upload-artifact@v1
-      with:
-        name: sycl_docs
-        path: sycl_docs.tar.gz
     - name: Deploy
       if: ${{ github.event_name == 'schedule' }}
       env:
@@ -64,7 +48,10 @@ jobs:
         cd $GITHUB_WORKSPACE/docs
         git rm -rf .
         touch .nojekyll
-        yes | \cp -rf $GITHUB_WORKSPACE/rendered/* .
+        yes | \cp -rf $GITHUB_WORKSPACE/build/tools/sycl/doc/html/* .
+        mv $GITHUB_WORKSPACE/build/tools/sycl/doc/doxygen/html doxygen/
+        mv $GITHUB_WORKSPACE/build/tools/clang/docs/html clang/
+        mv $GITHUB_WORKSPACE/build/tools/clang/docs/doxygen/html clang_doxygen/
         git config --global user.name "iclsrc"
         git config --global user.email "ia.compiler.tools.git@intel.com"
         git add .

--- a/.github/workflows/gh_pages.yml
+++ b/.github/workflows/gh_pages.yml
@@ -3,6 +3,13 @@ name: Generate Doxygen documentation
 on:
   schedule:
   - cron: 0 1 * * *
+  pull_request:
+    branches:
+    - sycl
+    paths:
+    - '.github/workflows/gh_pages.yml'
+    - 'clang/docs/**'
+    - 'sycl/docs/**'
 
 jobs:
   build:
@@ -16,7 +23,7 @@ jobs:
     - name: Install deps
       run: |
         sudo apt-get install -y doxygen graphviz ssh ninja-build
-        sudo pip3 install 'sphinx==4.1.2' 'myst-parser==0.15.1'
+        sudo pip3 install 'sphinx==4.1.2' 'myst-parser==0.15.1' 'recommonmark==0.7.1'
     - name: Build Docs
       run: |
         mkdir -p $GITHUB_WORKSPACE/build
@@ -27,7 +34,24 @@ jobs:
         cmake --build . --target doxygen-clang
         cmake --build . --target docs-sycl-html
         cmake --build . --target docs-clang-html
+        cd $GITHUB_WORKSPACE
+        mkdir rendered
+        cd rendered
+        yes | \cp -rf $GITHUB_WORKSPACE/build/tools/sycl/doc/html/* .
+        mv $GITHUB_WORKSPACE/build/tools/sycl/doc/doxygen/html doxygen/
+        mv $GITHUB_WORKSPACE/build/tools/clang/docs/html clang/
+        mv $GITHUB_WORKSPACE/build/tools/clang/docs/doxygen/html clang_doxygen/
+    - name: Pack
+      if: ${{ github.event_name != 'schedule' }}
+      run: tar -czf sycl_docs.tar.gz -C $GITHUB_WORKSPACE/rendered .
+    - name: Upload docs as artifact
+      if: ${{ github.event_name != 'schedule' }}
+      uses: actions/upload-artifact@v1
+      with:
+        name: sycl_docs
+        path: sycl_docs.tar.gz
     - name: Deploy
+      if: ${{ github.event_name == 'schedule' }}
       env:
         SSH_KEY: ${{secrets.ACTIONS_DEPLOY_KEY}}
       run: |
@@ -40,10 +64,7 @@ jobs:
         cd $GITHUB_WORKSPACE/docs
         git rm -rf .
         touch .nojekyll
-        yes | \cp -rf $GITHUB_WORKSPACE/build/tools/sycl/doc/html/* .
-        mv $GITHUB_WORKSPACE/build/tools/sycl/doc/doxygen/html doxygen/
-        mv $GITHUB_WORKSPACE/build/tools/clang/docs/html clang/
-        mv $GITHUB_WORKSPACE/build/tools/clang/docs/doxygen/html clang_doxygen/
+        yes | \cp -rf $GITHUB_WORKSPACE/rendered/* .
         git config --global user.name "iclsrc"
         git config --global user.email "ia.compiler.tools.git@intel.com"
         git add .

